app/controllers/medications_controller.rb

```diff
@@ -90,15 +97,26 @@ def destroy
 
   # Save refill date to Google calendar
   def save_refill_to_google_calendar(medication)
-    return unless current_user.google_oauth2_enabled? &&
-                  medication.add_to_google_cal == '1'
+    return true unless current_user.google_oauth2_enabled? &&
+                       medication.add_to_google_cal == '1'
```

Comment:
if I drop the string and do a simple comparison it fails, I think it's due to it being a temporary property, being attr_accessor on the model and passed from the form.

I haven't thought of a way to work around this yet, but the issue was raised somewhere else if I can recall about storing the google reminder state on the model and handling it more intelligently after initial creation. Would this perhaps require its own issue or do you want to rope it in here as well?